server/src/main/java/org/elasticsearch/action/StreamableResponseActionType.java

```diff
@@ -29,7 +29,7 @@
 public abstract class StreamableResponseActionType<Response extends ActionResponse> extends ActionType<Response> {
 
     protected StreamableResponseActionType(String name) {
-        super(name);
+        super(name, null);
```

Comment:
You can pass a lambda here, and then make getResponseReader final on ActionType

Reply:
I couldn't because of the dependency on `newResponse`, it could not be referenced in the lambda before the call to super